crates/ide-completion/src/context/analysis.rs

```diff
@@ -1140,7 +1143,52 @@ fn pattern_context_for(
                         })();
                         return (PatternRefutability::Irrefutable, has_type_ascription)
                     },
-                    ast::MatchArm(_) => PatternRefutability::Refutable,
+                    ast::MatchArm(match_arm) => {
+                       let missing_variants_opt = match_arm
+                            .syntax()
+                            .parent()
+                            .and_then(ast::MatchArmList::cast)
+                            .and_then(|match_arm_list| {
+                                match_arm_list
+                                .syntax()
+                                .parent()
+                                .and_then(ast::MatchExpr::cast)
```

Comment:
Not too relevant for the PR per se but as a note I feel like we should add extension methods here for fetching the parent arm list and parent match expr since those are always children of one another only.

Reply:
You mean adding extension methods on `ast::MatchArm`?